keps/sig-apps/3541-add-enforcedRollingUpdate-to-statefulset-update-strategies/README.md

```diff
@@ -0,0 +1,679 @@
+    // +optional
+    EnforcedRollingUpdate *RollingUpdateStatefulSetStrategy `json:"enforcedRollingUpdate,omitempty"`
+}
+```
+
+### Implementation Changes
+
+1. Before, when updating statefulSet, we should make sure that current replicas should be Running, Ready and Available.
+Now when StatefulSetUpdateStrategyType is EnforcedRollingUpdate, we'll continue the process.
+
+2. Before, when deleting so called condemned pods who are pods beyond the updated statefulSet replicas, we'll wait for all predecessors to be Running and Ready prior to attempting a deletion.
+Now when StatefulSetUpdateStrategyType is EnforcedRollingUpdate, we'll continue the process.
+
+When updating pods that doesn't match the update revision, we'll keep the logic and if the largest ordinal pod failed to turn to health, we'll not update the smaller ordinal pods.
+
+### Comparison with Existing Solutions
+
+| Solution                           | Sequential Ordering | Automatic Recovery | Behavior When Pod Stuck           | Use Case                               |
+| ---------------------------------- | ------------------- | ------------------ | --------------------------------- | -------------------------------------- |
+| `RollingUpdate`                    | Yes                 | No                 | Halts completely, waits forever   | Traditional stateful apps              |
+| `RollingUpdate` + `maxUnavailable` | Yes                 | No                 | **Still halts completely**        | Faster updates, but same stuck problem |
+| `OnDelete`                         | Yes                 | No                 | Fully manual control              | Maximum safety/control                 |
+| `Parallel` + `maxUnavailable`      | No                  | No                 | Halts, loses ordering             | Not suitable for ordered apps          |
+| `EnforcedRollingUpdate`            | Yes                 | Yes                | Automatically replaces stuck pods | Automated recovery needed              |
+
+`maxUnavailable` allows updating multiple pods simultaneously but does not change the fundamental behavior if any pod gets stuck, the entire update halts. The controller still waits for all pods to become Ready before proceeding.
+
+### Test Plan
+
+[x] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+##### Unit tests
+
+- `pkg/apis/apps/validation`: `2022-09-28` - `90.8%`
+- `pkg/controller/statefulset`: `2022-09-28` - `85.5%`
+- `pkg/registry/apps/statefulset`: `2022-09-28` - `66.7%`
+
+##### Integration tests
+
+We should cover below scenarios:
+
+- With update strategy set to `rollingUpdate`, broken statefulSet will not recover even applying with a fixed configuration.
```

Comment:
Throughout the entire document please use `StatefulSet` consistently, since that's the resource name.